lib/bolt/project.rb

```diff
@@ -25,8 +25,14 @@ def self.default_project
     # hierarchy, falling back to the default if we reach the root.
     def self.find_boltdir(dir)
       dir = Pathname.new(dir)
-      if (dir + BOLTDIR_NAME).directory?
-        new(dir + BOLTDIR_NAME, 'embedded')
+      # allows for any case of Boltdir, BoltDir, boltdir, etc
+      # note: we're sorting the directories here so that we consistently pick the
+      #       same Boltdir between runs, and we always choose the lexicographical "less"
+      #       Boltdir name rather than whatever the first name we run into as returned by
+      #       the filesystem
+      boltdirs = dir.children.sort.select { |c| c.directory? && c.basename.to_s.casecmp?(BOLTDIR_NAME) }
+      if !boltdirs.empty?
```

Comment:
We should issue a warning and say which one we're using if more than one is found.